src/chromo/models/pythia8.py

```diff
@@ -161,28 +173,61 @@ def _cross_section(self, kin=None):
             diffractive_axb=st.sigmaAXB,
         )
 
+        # TODO use pythia.info.sigmaGen(i) for cross-sections including nucleus in the
+        # initial collision system.
+        #   pythia.info.sigmaTot() are the estimated cross-section values for the
+        # equivalent pp collision for each sub-processes.
+        #   pythia.info.sigmaGen(i) are the cross-section values generated by the MC
+        # events for each subprocesses. Best is to grab these values at the end of the
+        # event generation when the full statistics is available.
+        # The subprocesses are defined with codes in Pythia:
+        #    i   subprocess
+
+    #   101  non-diffractive
+    #   102  A B -> A B elastic
+    #   103  A B -> X B single diffractive
+    #   104  A B -> A X single diffractive
+    #   105  A B -> X X double diffractive
+    #   106  A B -> A X B central diffractive
+
     def _set_kinematics(self, kin):
         config = self._config[:]
 
+        # Path prefix for some cache files that Pythia8 generates
+        # to speed up subsequent runs
+        cache_prefix = str(
+            _cache_base_dir()
+            / "Pythia8"
+            / f"cache_{int(kin.p1)}_{int(kin.p2)}_{kin.ecm / MeV:.0f}".replace("-", "m")
+        )
+
         # TODO use numpy PRNG instead of Pythia's
         config += [
             # use center-of-mass frame
             "Beams:frameType = 1",
             # do not print progress
             "Next:numberCount = 0",
+            "MultipartonInteractions:reuseInit = 3",
+            f"MultipartonInteractions:initFile = {cache_prefix}.mpi",
         ]
 
-        if (kin.p1.A or 0) > 1 or (kin.p2.A or 1) > 1:
-            import warnings
-
-            warnings.warn(
-                "Support for nuclei is experimental; event generation takes a long time",
-                RuntimeWarning,
-            )
+        # TODO Pythia8 likes to say this:
+        #  To avoid refitting, add the following lines to your configuration file:
+        #     HeavyIon:SigFitNGen = 0
+        #     HeavyIon:SigFitDefPar = 18.39,1.91,0.36
+        # but these numbers of collision-specific, so we cannot just set this globally.
+        # We could generate collision-specific cache files like we do for the
+        #  MPI stuff and set the numbers via those. To get the numbers, call
+        # SubCollisionModel::getParm(), which is complicated to call from the Pythia
+        # object, this is work in progress.
```

Comment:
I’ve seen some draft in your edited pybind interface? Do you know what exactly is the problem? Is the function exposed in the original pythia8 python interface?

Reply:
I forgot to answer this. This is the issue: It is not clear to me how to get the parameters 
`HeavyIon:SigFitDefPar = 18.39,1.91,0.36`
etc in an automatic way directly from Pythia. I don't want to scrape the text output, that's brittle and ugly. This is something to handle in a follow-up PR.

Reply:
Don't the cache files solve this problem? I saw a cross section fit file among the generated ones.